command_line_assistant/commands/README.md

```diff
@@ -0,0 +1,221 @@
+4. **Easy to understand**: Clear, linear code flow
+5. **Maintainable**: Reduced complexity makes the code easier to maintain and extend
+
+## Architecture Overview
+
+### Core Components
+
+- **`registry.py`**: Simple command registration system with decorators
+- **`utils.py`**: Common utilities and helpers for commands
+- **Individual command files**: `feedback.py`, `chat.py`, `history.py`, `shell.py`
+
+### Command Structure
+
+Each command is now implemented as a simple function with decorators:
+
+```python
+from command_line_assistant.utils.cli import argument, command
+from command_line_assistant.commands.utils import create_utils
+
+@command("mycommand", help="Description of my command")
+@argument("positional_arg", help="A positional argument")
+@argument("-f", "--flag", action="store_true", help="An optional flag")
+@argument("-v", "--value", type=str, help="An optional value")
+def my_command(args: Namespace, context: CommandContext) -> int:
+    """Command implementation."""
+    utils = create_utils(context, args)  # Pass args for plain support
+
+    try:
+        # Command logic here
+        utils.render_success("Command executed successfully!")
+        return 0
+    except Exception as e:
+        utils.render_error(str(e))
+        return 1
+```
+
+## Creating a New Command
+
+### Step 1: Create the command function
+
+```python
+@command("newcmd", help="My new command")
+@argument("message", help="Message to display")
+@argument("-u", "--uppercase", action="store_true", help="Uppercase the message")
+def new_command(args: Namespace, context: CommandContext) -> int:
+    utils = create_utils(context, args)  # Pass args for plain support
+
+    message = args.message.upper() if args.uppercase else args.message
+    utils.render_success(message)
+    return 0
+```
+
+### Step 2: Import in `__init__.py`
+
+Add your command module to the imports in `__init__.py`:
+
+```python
+from command_line_assistant.commands import feedback, chat_simple, newcmd
+```
+
+That's it! No factories, no complex registration, no multiple inheritance.
+
+## Available Utilities
+
+The `CommandUtils` class provides common functionality:
+
+### Renderers
+- `utils.render_success(message)` - Success messages
+- `utils.render_warning(message)` - Warning messages
+- `utils.render_error(message)` - Error messages
+- `utils.text_renderer` - Direct access to text renderer
+- `utils.warning_renderer` - Direct access to warning renderer
+- `utils.error_renderer` - Direct access to error renderer
+
+**Note**: All renderers automatically respect the `--plain` flag if present in args.
+
+### D-Bus Proxies
+- `utils.chat_proxy` - Chat interface proxy
+- `utils.history_proxy` - History interface proxy
+- `utils.user_proxy` - User interface proxy
+
+### Helper Methods
+- `utils.get_user_id()` - Get current user ID
+- `utils.context` - Access to command context
+
+## Migration from Old Structure
+
+### Old Structure (Complex)
```

Comment:
I think since this PR changes all of the commands, this migration example is not needed. Further migrations will not happen since they'll just be written the new way off the bat.